modules/cli-options/src/main/scala/scala/cli/commands/FmtOptions.scala

```diff
@@ -46,6 +46,9 @@ final case class FmtOptions(
     scalafmtArg: List[String] = Nil,
 
   @Group("Format")
+  @HelpMessage("Custom path to the directory with .scalafmt.conf file.")
+    configDir: Option[String] = None,
```

Comment:
why pass the path to the directory containing the file and not the file itself?
It'd make a lot more sense to be able to pass a path to any text file containing the configuration. 
Someone might even keep a couple of different configurations in one directory, to run for different setups.

Moreover, this should be as close to `scalafmt`'s native options as our API allows.

AFAIK there are 2 options allowing to pass the configuration from a custom location to `scalafmt` itself:

- the `--config` (or `-c`) option, which allows to pass exactly what I mentioned - a path to any file (and not its surrounding directory
- the ` --config-str` option, which allows to pass a configuration as a string from the command line (similarly to how `scala-cli` allows to pass snippets)

(more on that in [scalafmt's own docs](https://scalameta.org/scalafmt/docs/installation.html)

Now, we probably should still rename these, as they will be too ambigous in `scala-cli` context.
Some suggestions:
- `--config` could become `--scalafmt-conf` (maybe with an alias for `--scalafmt-config`)
- `--config-str` similarly could be `--scalafmt-conf-str `(with aliases like `--scalafmt-config-str` and `--scalafmt-conf-snippet`)

Reply:
(support for an equivalent of `--config-str` is out of scope of the ticket btw, but since you're touching this anyway... 😉 )

Reply:
What if `--scalafmt-config-str` doesn't have `version` or `dialect` specified. Should `fmt` add it under the hood or throw an error?

Reply:
add it under the hood, definitely.
whatever would be passed with `--scalafmt-conf-str` should be treated the same as any other configuration file.